From bce367a2ca60e37266a3f413c1a1c4db368479c2 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Wed, 3 Apr 2019 13:01:56 -0700
Subject: [PATCH 0608/1852] v4l: xilinx: scd: Clean up #include statements

commit a041561daa922710217b5f390707da0398afc211 from
https://github.com/Xilinx/linux-xlnx.git

xilinx-scenechange.h contains various #include statements not needed by
the header itself but by the multiple .c files that include it. Move
them to the .c files and remove unneeded includes.

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../xilinx/xilinx-scenechange-channel.c       | 15 +-----------
 .../platform/xilinx/xilinx-scenechange-dma.c  | 16 ++++---------
 .../platform/xilinx/xilinx-scenechange.c      |  7 ++++++
 .../platform/xilinx/xilinx-scenechange.h      | 23 ++++++++-----------
 4 files changed, 22 insertions(+), 39 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-scenechange-channel.c b/drivers/media/platform/xilinx/xilinx-scenechange-channel.c
index 2161de7de0ab..989ec6e44b43 100644
--- a/drivers/media/platform/xilinx/xilinx-scenechange-channel.c
+++ b/drivers/media/platform/xilinx/xilinx-scenechange-channel.c
@@ -8,23 +8,10 @@
  *          Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
  */
 
-#include <linux/clk.h>
-#include <linux/delay.h>
-#include <linux/dmaengine.h>
-#include <linux/module.h>
-#include <linux/gpio/consumer.h>
-#include <linux/interrupt.h>
-#include <linux/kernel.h>
-#include <linux/module.h>
 #include <linux/of.h>
-#include <linux/of_irq.h>
-#include <linux/platform_device.h>
-#include <linux/slab.h>
-#include <linux/xilinx-v4l2-controls.h>
 #include <linux/xilinx-v4l2-events.h>
+
 #include <media/v4l2-async.h>
-#include <media/v4l2-ctrls.h>
-#include <media/v4l2-device.h>
 #include <media/v4l2-event.h>
 #include <media/v4l2-subdev.h>
 
diff --git a/drivers/media/platform/xilinx/xilinx-scenechange-dma.c b/drivers/media/platform/xilinx/xilinx-scenechange-dma.c
index 36b90e9d7788..6f44744b8e1e 100644
--- a/drivers/media/platform/xilinx/xilinx-scenechange-dma.c
+++ b/drivers/media/platform/xilinx/xilinx-scenechange-dma.c
@@ -8,22 +8,14 @@
  *          Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
  */
 
-#include <linux/bitops.h>
-#include <linux/clk.h>
-#include <linux/dmaengine.h>
+#include <linux/delay.h>
 #include <linux/dma-mapping.h>
-#include <linux/init.h>
-#include <linux/interrupt.h>
-#include <linux/io.h>
-#include <linux/io-64-nonatomic-lo-hi.h>
-#include <linux/iopoll.h>
-#include <linux/module.h>
-#include <linux/of_address.h>
+#include <linux/dmaengine.h>
 #include <linux/of_dma.h>
-#include <linux/of_irq.h>
-#include <linux/platform_device.h>
 #include <linux/slab.h>
 
+#include "../../../dma/dmaengine.h"
+
 #include "xilinx-scenechange.h"
 
 /**
diff --git a/drivers/media/platform/xilinx/xilinx-scenechange.c b/drivers/media/platform/xilinx/xilinx-scenechange.c
index b1d7daed8dd7..71297e19c981 100644
--- a/drivers/media/platform/xilinx/xilinx-scenechange.c
+++ b/drivers/media/platform/xilinx/xilinx-scenechange.c
@@ -8,6 +8,13 @@
  *          Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
  */
 
+#include <linux/clk.h>
+#include <linux/gpio/consumer.h>
+#include <linux/interrupt.h>
+#include <linux/module.h>
+#include <linux/of.h>
+#include <linux/platform_device.h>
+
 #include "xilinx-scenechange.h"
 
 #define XSCD_RESET_DEASSERT	(0)
diff --git a/drivers/media/platform/xilinx/xilinx-scenechange.h b/drivers/media/platform/xilinx/xilinx-scenechange.h
index e19a497b70ca..9bbe3814aa8b 100644
--- a/drivers/media/platform/xilinx/xilinx-scenechange.h
+++ b/drivers/media/platform/xilinx/xilinx-scenechange.h
@@ -11,22 +11,19 @@
 #ifndef _XILINX_SCENECHANGE_H_
 #define _XILINX_SCENECHANGE_H_
 
-#include <linux/clk.h>
-#include <linux/delay.h>
+#include <linux/bitops.h>
 #include <linux/dmaengine.h>
-#include <linux/gpio/consumer.h>
 #include <linux/interrupt.h>
-#include <linux/kernel.h>
-#include <linux/module.h>
-#include <linux/of.h>
-#include <linux/platform_device.h>
-#include <linux/xilinx-v4l2-controls.h>
-
-#include <media/v4l2-async.h>
-#include <media/v4l2-ctrls.h>
-#include <media/v4l2-device.h>
+#include <linux/io.h>
+#include <linux/mutex.h>
+#include <linux/spinlock.h>
+
 #include <media/v4l2-subdev.h>
-#include "../../../dma/dmaengine.h"
+
+struct clk;
+struct device;
+struct device_node;
+struct gpio_desc;
 
 /* Register/Descriptor Offsets */
 #define XSCD_CTRL_OFFSET		0x000
-- 
2.31.1

